### PR TITLE
feature/1: initial implementation

### LINKED ISSUE #1: Create initial implementation
Create initial implementation

### PR DESCRIPTION
**Pull request by jakobbraun**
Closes #1

### CODE REVIEW

File: LICENSE [hunk 1]
```diff
@@ -0,0 +1,21 @@
+MIT License
+
+Copyright (c) 2020 Exasol
```
Comment (morazow) [inline LICENSE]:
```suggestion
Copyright (c) 2021 Exasol
```

File: README.md [hunk 2]
```diff
@@ -0,0 +1,142 @@
+# Autogenerated Resource Verifier for Java
+
+[![Maven Central](https://img.shields.io/maven-central/v/com.exasol/autogenerated-resource-verifier-java)](https://search.maven.org/artifact/com.exasol/autogenerated-resource-verifier-java)
+
+This Java library allows you to define autogenerated resources in Java.
```
Comment (morazow) [inline README.md]:
```suggestion
This Java library allows you to verify autogenerated resources in Java.
```

Comment (morazow) [reply README.md]: Did you meant `verify` here? If so, maybe add one more sentence what we are verifying.

File: README.md [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+                </systemProperties>
+            </configuration>
+        </execution>
+    </executions>
+</plugin>
+```
+
+Now you can run the validation using `mvn package`.
+
+The verifier can also fix (override) the generated resource. To do so, run:
```
Comment (morazow) [inline README.md]:
```suggestion
The verifier can also fix the generated resource. To do so, run:
```

File: README.md [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+
+```shell
+mvn package -DfixAutogeneratedResources=true
+```
+
+### Run Time Dependencies
+
+Running the Virtual Schema requires a Java Runtime version 11 or later.
```
Comment (morazow) [inline README.md]:
```suggestion
Running the project requires a Java Runtime version 11 or later.
```

File: README.md [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+
+| Dependency                                                         | Purpose                                                | License                       |
+|--------------------------------------------------------------------|--------------------------------------------------------|-------------------------------|
+| [Exasol Error Reporting][error-reporting-java]                     | Common JDBC functions for Virtual Schemas adapters     | MIT License                   |
```
Comment (morazow) [inline README.md]:
```suggestion
| [Exasol Error Reporting][error-reporting-java]                     | Java builders for Exasol error messages.     | MIT License                   |
```

File: README.md [hunk 6]
```diff
@@ -0,0 +1,142 @@
...
+[exasol-testcontainers]: https://github.com/exasol/exasol-testcontainers
+
+[user-guide]: https://docs.exasol.com/database_concepts/virtual_schemas.htm
+
+[virtual-schemas]: https://github.com/exasol/virtual-schemas
+
+[vs-api]: https://github.com/exasol/virtual-schema-common-java/blob/master/doc/development/api/virtual_schema_api.md
+
+[vs-doc]: https://github.com/exasol/virtual-schemas/tree/master/doc
+
```
Comment (morazow) [inline README.md]:
```suggestion
```

File: doc/changes/changes_0.1.0.md [hunk 7]
```diff
@@ -0,0 +1,14 @@
+#Autogenerated resource verifier 0.1.0, released 2021-02-11
```
Comment (morazow) [inline doc/changes/changes_0.1.0.md]: Please do not forget to update the release date.

File: doc/changes/changes_0.1.0.md [hunk 8]
```diff
@@ -0,0 +1,14 @@
...
+
+## Code name: Initial Release
+
+## Features / Enhancements
+
+* #1: Initial Implementation
+
+## Dependency Updates:
+
```
Comment (morazow) [inline doc/changes/changes_0.1.0.md:10]:
```suggestion

* Added `com.exasol:error-reporting-java:0.2.2`
```

File: src/main/java/com/exasol/autogeneratedresourceverifier/AutogeneratedResourceVerifier.java [hunk 9]
```diff
@@ -0,0 +1,100 @@
...
+        }
+    }
+
+    private Path readProjectDir() {
+        final String projectDirString = System.getProperty(PROJECT_DIR_SYSTEM_PROPERTY, "");
+        if (projectDirString.equals("")) {
+            throw new IllegalStateException(ExaError.messageBuilder("E-ARVJ-6")
+                    .message("Missing property " + PROJECT_DIR_SYSTEM_PROPERTY + ".")
+                    .mitigation(
+                            "Please set the property to the absolut path ro your projects root. Usually this is done using the exec-maven-plugin.")
```
Comment (morazow) [inline src/main/java/com/exasol/autogeneratedresourceverifier/AutogeneratedResourceVerifier.java]:
```suggestion
                            "Please set the property to the absolute path of your projects root. Usually this is done using the exec-maven-plugin.")
```

File: src/main/java/com/exasol/autogeneratedresourceverifier/AutogeneratedResourceVerifier.java [hunk 10]
```diff
@@ -0,0 +1,100 @@
...
+    }
+
+    private void validateFile(final String expected, final Path resourcePath) {
+        assertResourceExists(resourcePath);
+        final String actual = readResource(resourcePath);
+        if (!expected.trim().equals(actual.trim())) {
+            throw new IllegalStateException(ExaError.messageBuilder("E-ARVJ-4")
+                    .message("Autogenerated resource {{resource name}} is outdated.")
+                    .parameter("resource name", resourcePath.toString())
+                    .mitigation("You can enable automatically update by adding -D" + FIX_SYSTEM_PROPERTY + "=true.")
```
Comment (morazow) [inline src/main/java/com/exasol/autogeneratedresourceverifier/AutogeneratedResourceVerifier.java]:
```suggestion
                    .mitigation("You can enable automatic update by adding -D" + FIX_SYSTEM_PROPERTY + "=true.")
```

File: src/main/java/com/exasol/autogeneratedresourceverifier/AutogeneratedResourceVerifier.java [hunk 11]
```diff
@@ -0,0 +1,100 @@
...
+                    .parameter("resource name", resourcePath.toString()).toString(), exception);
+        }
+    }
+
+    private void createEnclosingDirectoryIfNotExists(final Path resourcePath) {
+        final File directory = resourcePath.toFile().getParentFile();
+        if (!directory.exists()) {
+            if (!directory.mkdirs()) {
+                throw new IllegalStateException(ExaError.messageBuilder("E-ARVJ-3")
+                        .message("Failed to crate directory {{directory name}} for autogenerated resource.")
```
Comment (morazow) [inline src/main/java/com/exasol/autogeneratedresourceverifier/AutogeneratedResourceVerifier.java]:
```suggestion
                        .message("Failed to create directory {{directory name}} for autogenerated resource.")
```

File: src/main/java/com/exasol/autogeneratedresourceverifier/AutogeneratedResourceVerifier.java [hunk 12]
```diff
@@ -0,0 +1,100 @@
...
+    }
+
+    private boolean isFixEnabled() {
+        return System.getProperty(FIX_SYSTEM_PROPERTY, "").equals("true");
+    }
+
+    private void assertResourceExists(final Path resourcePath) {
+        if (!resourcePath.toFile().exists()) {
+            throw new IllegalStateException(
+                    ExaError.messageBuilder("E-ARVJ-2").message("Autogenerated resource {{resource name}} missing.")
```
Comment (morazow) [inline src/main/java/com/exasol/autogeneratedresourceverifier/AutogeneratedResourceVerifier.java]:
```suggestion
                    ExaError.messageBuilder("E-ARVJ-2").message("Autogenerated resource {{resource name}} is missing.")
```

File: src/main/java/com/exasol/autogeneratedresourceverifier/AutogeneratedResourceVerifier.java [hunk 13]
```diff
@@ -0,0 +1,100 @@
...
+
+/**
+ * This class verifies that an {@link AutogeneratedResource} has the expected value. In addition it can fix it.
+ */
+public class AutogeneratedResourceVerifier {
+    public static final String FIX_SYSTEM_PROPERTY = "fixAutogeneratedResources";
+    public static final String PROJECT_DIR_SYSTEM_PROPERTY = "projectDir";
+
+    /**
+     * Verify that an {@link AutogeneratedResource} has the expected value
```
Comment (morazow) [inline src/main/java/com/exasol/autogeneratedresourceverifier/AutogeneratedResourceVerifier.java]:
```suggestion
     * Verify that an {@link AutogeneratedResource} has the expected value.
```

File: README.md [hunk 14]
```diff
@@ -0,0 +1,142 @@
...
+## Usage
+
+Define the content of your document in your test source code:
+
+```java
+public class MyReportGenerator {
+
+    public static void main(final String[] args) {
+        new AutogeneratedResourceVerifier().verifyResource(Path.of(args[0]), new CapabilitiesReport());
```
Comment (morazow) [inline README.md]:
```suggestion
        new AutogeneratedResourceVerifier().verifyResource(Path.of(args[0]), new MyReport());
```

File: README.md (hunk 2 above)
Comment (morazow) [reply README.md]: After look the code, maybe `create` is better here.

```
This Java library allows you to automatically create resources in Java.
```

**Review by morazow — CHANGES_REQUESTED** [review]

File: README.md (hunk 2 above)
Comment (jakobbraun) [reply README.md]:
```suggestion
This Java library allows you to automatically create resources in Java.
```

File: doc/changes/changes_0.1.0.md (hunk 7 above)
Comment (jakobbraun) [reply doc/changes/changes_0.1.0.md]:
```suggestion
#Autogenerated resource verifier 0.1.0, released 2021-02-12
```

**Review by morazow — APPROVED** [review]

**Pull request merged**